mandatory/3-playing-computer.js

```diff
@@ -7,36 +7,47 @@
   Answer the following questions:
 
   1. This program throws an error. Why? (If you can't find it, try executing it).
-  2. Remove the line that throws the error.
+    [ console.log(b) throws error as b is undefined]
+  2. Remove the line that throws the error. 
+    [commented out the line causing errors]
   3. What is printed to the console?
+    [2, 6, 6, 4, 7, 9, 8, 6, 9, 13, 10, 8]
   4. How many times is "f1" called?
+    [10 times, used console.log("find", a)]
   5. How many times is "f2" called?
+    [2 times, used console.log("search", x)]
   6. What value does the "a" parameter take in the first "f1" call?
+    [value a = 9, used console.log("before", a)]
```

Comment:
I think you've slightly misunderstood the question. It's asking for the 'a' parameter. As in the first parameter of the f1 function. This is passed in as the variable 'i' when the f1 function is called in the for loop. As the f1 function is called when the index is first an odd number on its second iteration of the loop, the index is incremented once to make it 1.